File: articles/the-history-of-data-structures-and-algorithm-analysis/fortran/growth_rates.f90
program growth_rates
  implicit none
  integer :: i
  real :: ns(4), n
  ns = (/ 10.0, 100.0, 1000.0, 10000.0 /)
  do i = 1, 4
    n = ns(i)
    print *, "n=", n, " log2=", log(n)/log(2.0), " nlogn=", n*log(n)/log(2.0), " n2=", n*n
  end do
end program growth_rates

File: articles/the-history-of-data-structures-and-algorithm-analysis/prolog/growth_rates.pl
growth_rate(N, Log2, NLogN, N2) :-
    Log2 is log(N) / log(2),
    NLogN is N * Log2,
    N2 is N * N.

% Query example:
% ?- growth_rate(1000, Log2, NLogN, N2).
